specs/enable_gluster_native_events.adoc

```diff
@@ -0,0 +1,147 @@
+= Enable gluster native events in tendrl
+
+Events generated by glusterfs events API should be exposed by tendrl.
+
+
+== Problem description
+
+Currently tendrl is not utilizing the events exposed by glusterfs events API.
+Tendrl should be enhanced to consume and handle the events.
+
+== Use Cases
+
+* Glusterfs events API pushes events for certain internal glusterfs failure conditions
+which is not track able by tendrl. Tendrl can raise suitable alerts for such events.
+
+* The interval of sds state sync can be increased (to 30 min) and we should have event
+driven state sync. which will considerably increase the efficiency of gluster integration
+service.
+
+== Proposed change
+
+The event receiver web-hook can be run on all the gluster storage nodes as a
+thread/greenlet in gluster-integration daemon. And we should configure the
+gluster eventing daemon to push events in localhost mode (the events generated
+by gluster eventing daemon will be pushed to the web-hook running only on
+that particular node). With this we are eliminating single point of failure
+w.r.t event listener.
+
+Since some of the events are raised from all the nodes or raised multiple times
+(till alert condition gets fixed), tendrl should make sure that duplicate events
+are handled appropriately by checking the state information stored in etcd.
+
+In future, gluster integration state sync should be based on events received
+by glusterfs events API. Thus we make sure that we do the state sync only when
+there is a state change. Also we can have a 30 min sync, to make sure that we
+have a valid state in case events are missed.
+
+=== Alternatives
+
+An alternate approach to listen events is to have listener enabled in only one of
+the gluster-integration process, and configure all gluster events API to push
+events to this instance. Though this has slight advantage of not having to run the
+event listeners on all the nodes.
+
+The main problem with this approach is handling the failure of this listener node.
```

Comment:
My earlier comment asks about whether any part of the gluster-integration state sync, as currently implemented, is NOT node specific. If the state sync for each of the integration instances is independent of the others, it's better to go with the original approach rather than this alternative one.